python/mpt/tests/src/mpt/test_utils.py

```diff
@@ -74,3 +84,122 @@ def test_check_branch_node_invalid_x_branch(
+    ):
+        sorted_data = sorted(
+            data, key=lambda x: int.from_bytes(x.key, "little"), reverse=True
+        )
+        cairo_data = cairo_run("sort_account_diff", data)
+        assert cairo_data == sorted_data
+
+    @given(data=list_address_account_node_diff_entry_strategy)
+    def test_sort_account_diff_should_fail_if_not_descending_order(
+        self, cairo_programs, cairo_run_py, data: List[AddressAccountNodeDiffEntry]
+    ):
+        with patch_hint(
+            cairo_programs,
+            "sort_account_diff",
+            """
+# Extract the list of pointers directly
+pointers = [memory[ids.diffs_ptr.address_ + i] for i in range(ids.diffs_len)]
+
+# Sort pointers based on the key values they point to, in descending order
+sorted_pointers = sorted(pointers, key=lambda ptr: memory[ptr], reverse=True)
+
+# Invert the order of the last two elements
+sorted_pointers[-2], sorted_pointers[-1] = sorted_pointers[-1], sorted_pointers[-2]
+segments.load_data(ids.buffer, sorted_pointers)
+
+indices = list(range(ids.diffs_len))
+sorted_to_original_index_map = sorted(indices, key=lambda i: memory[pointers[i]], reverse=True)
+sorted_to_original_index_map[-2], sorted_to_original_index_map[-1] = sorted_to_original_index_map[-1], sorted_to_original_index_map[-2]
+segments.load_data(ids.sorted_to_original_index_map, sorted_to_original_index_map)
+            """,
+        ):
+            with strict_raises(
+                Exception, match="ValueError: Array is not sorted in descending order"
+            ):
+                cairo_run_py("sort_account_diff", data)
+
+    @given(data=list_address_account_node_diff_entry_strategy)
+    def test_sort_account_diff_different_lists(
+        self, cairo_programs, cairo_run_py, data: List[AddressAccountNodeDiffEntry]
+    ):
+        with patch_hint(
+            cairo_programs,
+            "sort_account_diff",
+            """
+# Extract the list of pointers directly
+pointers = [memory[ids.diffs_ptr.address_ + i] for i in range(ids.diffs_len)]
+sorted_pointers = sorted(pointers, key=lambda ptr: memory[ptr], reverse=True)
+
+# BAD HINT: not a permutation of the input list
+sorted_pointers[0] += 1
+
+segments.load_data(ids.buffer, sorted_pointers)
+
+indices = list(range(ids.diffs_len))
+sorted_to_original_index_map = sorted(indices, key=lambda i: memory[pointers[i]], reverse=True)
+segments.load_data(ids.sorted_to_original_index_map, sorted_to_original_index_map)
+            """,
+        ):
+            with strict_raises(
+                Exception,
+                match="ValueError: Sorted element does not match original element at hint index",
+            ):
+                cairo_run_py("sort_account_diff", data)
+
+    @given(data=list_address_account_node_diff_entry_strategy)
+    def test_sort_account_diff_sorted_list_with_duplicates(
+        self, cairo_programs, cairo_run_py, data: List[AddressAccountNodeDiffEntry]
+    ):
+        with patch_hint(
```

Comment:
no need for patch hint in this test. you need to change the strategy to have duplicates

Reply:
up